Content.Client/Weapons/Ranged/Systems/GunSystem.cs

```diff
@@ -99,15 +99,11 @@ private void OnHitscan(HitscanEvent ev)
             if (a.Sprite is not SpriteSpecifier.Rsi rsi)
                 continue;
 
-            var coords = GetCoordinates(a.coordinates);
-
-            if (Deleted(coords.EntityId))
-                continue;
-
-            var ent = Spawn(HitscanProto, coords);
+            // we intentonally use this specific `Spawn` overload since it deals with grid
+            // rotation for us. we have to make sure grid rotation is accounted for for each
+            // *indivisual* sprite segment of the laser effect, 
+            var ent = Spawn(HitscanProto, TransformSystem.ToMapCoordinates(GetCoordinates(a.coordinates)), rotation: a.angle);
```

Comment:
Spawning it on mapcoords is still going to have the same problem of not handling shuttle movement.

Reply:
i've tried multiple ways of doing this, but it seems `SharedGridTraversalSystem` just reparents it back to the map. short of adding a way to bypass it, i'm not sure how to keep the parent to the source grid

Reply:
okay well i put in the bypass and everything works but now scary engine PR